File: 프로그래머스/lv1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.js
function solution(s) {
    const arr = ["zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"];
    let answer = "";
    let key = "";
    for(let i = 0; i < s.length; i++) {
        if(s.charAt(i) >= '0' && s.charAt(i) <= '9') {
            answer += s.charAt(i);
        } else {
            key += s.charAt(i);
            if (arr.includes(key)) {
                answer += arr.indexOf(key);
                key = "";
            }
        }
    }
    return Number(answer);
}

File: 프로그래머스/lv1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int solution(String s) {
        String[] numbers = {"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};

        Map<String, Integer> map = new HashMap<>();
        for (int i = 0; i < numbers.length; i++) {
            map.put(numbers[i], i);
        }

        StringBuilder br = new StringBuilder();
        char[] chars = s.toCharArray();
        String key = "";
        for (char c : chars) {
            if (c > 47 && c < 58) {
                br.append(c);
            } else {
                key += c;
                if (map.containsKey(key)) {
                    br.append(map.get(key));
                    key = "";
                }
            }
        }
        return Integer.parseInt(br.toString());
    }
}